atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientImpl.java

```diff
@@ -75,16 +98,48 @@ public CassandraClientImpl(Cassandra.Client client) {
             KeyRange range,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        ColumnParent colFam = getColumnParent(tableRef);
-        return client.get_range_slices(colFam, predicate, range, consistency_level);
+        int numberOfKeys = predicate.slice_range.count;
+        int numberOfColumns = range.count;
+
+        try (CloseableTrace trace = startLocalTrace(
+                "client.get_range_slices(table {}, number of keys {}, number of columns {}, consistency {}) on kvs.{}",
+                LoggingArgs.safeTableOrPlaceholder(tableRef),
+                numberOfKeys, numberOfColumns, consistency_level, kvsMethodName)) {
+            ColumnParent colFam = getColumnParent(tableRef);
+
+            return KvsProfilingLogger.maybeLog(
+                    (KvsProfilingLogger.CallableCheckedException<List<KeySlice>, TException>)
+                            () -> client.get_range_slices(colFam, predicate, range, consistency_level),
+                    (logger, timer) -> logger.log("client.get_range_slices({}, {}, {}, {}) on kvs.{}",
+                            LoggingArgs.tableRef(tableRef),
+                            SafeArg.of("number of keys", numberOfKeys),
+                            SafeArg.of("number of columns", numberOfColumns),
+                            SafeArg.of("consistency", consistency_level.toString()),
+                            SafeArg.of("kvsMethodName", kvsMethodName)));
+        }
     }
 
     @Override
     public void batch_mutate(String kvsMethodName,
             Map<ByteBuffer, Map<String, List<Mutation>>> mutation_map,
             ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
-        client.batch_mutate(mutation_map, consistency_level);
+        int numberOfMutations = mutation_map.size();
```

Comment:
this is slightly inaccurate, it's just the number of keys. number of mutations would be the sum of the sizes of the `List<Mutations>`s, I think

Reply:
I think you're correct, and I think that the next step here is to break the API again to include more information — e.g. the name of the tables that are being modified. I think I'm going to do that in a separate, since this PR seems to be big already.